src/averageAges.js

```diff
@@ -38,6 +51,17 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+
+  const callbackAge = (person) => person.died - person.born;
+  const callbackSum = (person1, person2) => person1 + person2;
+
+  const women = people.filter(person =>
+    withChildren
+      ? people.find(child => person.name === child.mother)
```

Comment:
Now in each iteration filter method we iterating through all people array.
We can avoid O**2 complaxity. Suggest before filter create object mothers (use reduce method) where key will be mother's name. Value can be any (list of children, true). We just need list of mothers.
```
const mothers = people.reduce((acc, person) => {
    acc[person.mother] = true;

    return acc;
  }, {});
  ```
And instead find check whether person is mother.
```
? mothers[person.name]
```